doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
+  new module repositories.
+
+  .. note::
+
+     It is not required to modify the names of existing module repositories
+     to make them comply with the above naming convention.
+
+* If the module has an external (upstream) project repository, the module
+  repository should preserve the upstream repository folder structure.
+
+* Modules should use "zephyr" as the default name for the repository main
+  branch. Branches for specific purposes, for example, a module branch for
+  an LTS Zephyr version, shall have names starting with the 'zephyr\_' prefix.
+
+  .. note::
+
+     It is **not** recommended in module repositories to have a 'master'
+     branch mirroring the master branch of the external repository.
+
+.. _modules_synchronization:
+
+Synchronizing with upstream
+===========================
+
+Requirements for allowed practices
+----------------------------------
+
+Any changes to module repositories, as a result of synchronizing with,
+upstream shall be brought in by pull requests. These pull requests
+shall be merg-able (i.e. with Rebase & Merge option, using Github UI).
+This ensures that a linear history is preserved in the module repositories.
+It also allows to easily review the incoming changea and run Zephyr CI,
+git lint, identity, and license checks directly on the set of changes
+that are to be brought into the module.
+
+.. note::
+
+     Force-pushing to a module's main branch is not allowed.
+
+Allowed practices
+---------------------
+
+The following practices are allowed and should be followed in all modules.
+
+**Updating modules with a diff from upstream:**
+Upstream changes brought as a single commit (manual diff) in a PR against
+the module's main branch. This approach is simple and should be applicable
+to all modules with the downside of supressing the upstream history in the
+module repository.
+
+The commit message is expected to identify the upstream repository, the
+version to which the module is updated (upstream version, tag, commit,
+etc.), and the reason for the doing the update.
+
+**Cherry picking upstream commits:**
+Upstream changes brought in by cherry-picking the upstream commits and
+submitting a PR with these commits against the module main branch. This
+approach is applicable to modules with an upstream project Git repository.
+The main advantages of this approach are the following
+
+* the upstream repository history may be preserved in the module repository,
+  if the cherry picked commits have a pointer to the original upstream commit
+* downstream module patches are less likely to be unintentionally overwritten
+  by the synchronization process.
+
+
+Contributing to Zephyr modules
+******************************
+
+.. _modules_contributing:
+
+
+Individual Roles & Responsibilities
+===================================
+
+To facilitate management of Zephyr module repositories, the following
+individual roles are defined.
+
+**Administration:** Each Zephyr module shall have an administrator
```

Comment:
This and the 'code ownership' roles, seem not to follow the MAINTAINERS.yml terminology and responsibilities, is that on purpose or should they not be similar?

Reply:
I infer that this is a deviation based on the granularity of github permissions.  I don't think every module maintainer should be granted github administrator rights to all modules in `zephyrproject-rtos`.  I do think it would be best that the module maintainer be given administrator rights to the modules they maintain, which include the ability to temporarily remove any hooks that prevent (non-force) push directly to the `zephyr` branch rather than go through the UI (if the UI would do a rebase or add a new merge commit or other undesirable actions).  But that may not be possible (and may not be acceptable).

Reply:
>  I do think it would be best that the module maintainer be given administrator rights to the modules they maintain, which include the ability to temporarily remove any hooks that prevent (non-force) push directly to the zephyr branch rather than go through the UI (if the UI would do a rebase or add a new merge commit or other undesirable actions). But that may not be possible (and may not be acceptable).

I'm agree with the above. Better to separate responsibilities precisely.
Also I'm using direct pushing for synchronizing zephyr MCUBoot repository with its upstream

Reply:
This was discussed in the meeting. The decision is to not give Admin or Merge rights to Module owners. 
For Admin, the reasoning is that the work could be done by the current Admins of the organization (it's not a lot of work anyways).
For Merge, the reasoning was that it does not help merging to the module branch, if one cannot update the Zephyr manifest.

Reply:
@mbolivar-nordic will apply the propose change.